src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ResourceBundle\Controller;
+
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use Hateoas\Representation\CollectionRepresentation;
+use Sulu\Bundle\ResourceBundle\Api\Filter;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\FilterDependencyNotFoundException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\FilterNotFoundException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\MissingFilterAttributeException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\MissingFilterException;
+use Sulu\Bundle\ResourceBundle\Filter\FilterManagerInterface;
+use Sulu\Component\Rest\Exception\EntityNotFoundException;
+use Sulu\Component\Rest\Exception\MissingArgumentException;
+use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactory;
+use Sulu\Component\Rest\ListBuilder\ListRepresentation;
+use Sulu\Component\Rest\RestController;
+use Sulu\Component\Rest\RestHelperInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class FilterController
+ * @package Sulu\Bundle\ResourceBundle\Controller
+ */
+class FilterController extends RestController implements ClassResourceInterface
+{
+    protected static $entityName = 'SuluResourceBundle:Filter';
+
+    protected static $entityKey = 'filters';
+
+    /**
+     * Retrieves a filter by id
+     *
+     * @param Request $request
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function getAction(Request $request, $id)
+    {
+        $locale = $this->getLocale($request);
+        $view = $this->responseGetById(
+            $id,
+            function ($id) use ($locale) {
+                /** @var Filter $filter */
```

Comment:
do you need that really? if you add a return phpdoc that it should be obvious! also you can remove the variable and return it immediatly, because it is not use